### ISSUE TITLE
Example code not working - MPG example

### ISSUE THREAD

**Issue by KirkDCO** [keras-team/autokeras#1163]
### Bug Description
Trying to get started using AutoKeras and finding that most of the example code does not work.  

### Bug Reproduction
Running the example here:  https://autokeras.com/tutorial/structured_data_regression/

### Setup Details
Include the details about the versions of:
 - OS type and version:  MacOS Catalina 10.15.4
 - Python: 3.8
 - autokeras: master (pulled 20.06.01)
 - keras-tuner: 1.0.1
 - scikit-learn: 0.23.1
 - numpy: 1.18.4
 - pandas: 1.0.4
 - tensorflow: 2.2.0

### Error

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-12-3dda062a9e40> in <module>
      5 # Evaluate the accuracy of the found model.
      6 print('Accuracy: {accuracy}'.format(
----> 7     accuracy=regressor.evaluate(x=test_dataset.drop(columns=['MPG']), y=test_dataset['MPG'])))

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/autokeras-1.0.3-py3.8.egg/autokeras/tasks/structured_data.py in evaluate(self, x, y, batch_size, **kwargs)
    133         if isinstance(x, str):
    134             x, y = self._read_from_csv(x, y)
--> 135         return super().evaluate(x=x,
    136                                 y=y,
    137                                 batch_size=batch_size,

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/autokeras-1.0.3-py3.8.egg/autokeras/auto_model.py in evaluate(self, x, y, **kwargs)
    443         """
    444         dataset = self._process_xy(x, y, False)
--> 445         return self.tuner.get_best_model().evaluate(x=dataset, **kwargs)
    446 
    447     def export_model(self):

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/autokeras-1.0.3-py3.8.egg/autokeras/engine/tuner.py in get_best_model(self)
     43 
     44     def get_best_model(self):
---> 45         model = super().get_best_models()[0]
     46         model.load_weights(self.best_model_path)
     47         return model

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/kerastuner/engine/tuner.py in get_best_models(self, num_models)
    229         """
    230         # Method only exists in this class for the docstring override.
--> 231         return super(Tuner, self).get_best_models(num_models)
    232 
    233     def _deepcopy_callbacks(self, callbacks):

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/kerastuner/engine/base_tuner.py in get_best_models(self, num_models)
    236         """
    237         best_trials = self.oracle.get_best_trials(num_models)
--> 238         models = [self.load_model(trial) for trial in best_trials]
    239         return models
    240 

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/kerastuner/engine/base_tuner.py in <listcomp>(.0)
    236         """
    237         best_trials = self.oracle.get_best_trials(num_models)
--> 238         models = [self.load_model(trial) for trial in best_trials]
    239         return models
    240 

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/kerastuner/engine/tuner.py in load_model(self, trial)
    154         best_epoch = trial.best_step
    155         with hm_module.maybe_distribute(self.distribution_strategy):
--> 156             model.load_weights(self._get_checkpoint_fname(
    157                 trial.trial_id, best_epoch))
    158         return model

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py in load_weights(self, filepath, by_name, skip_mismatch)
    248         raise ValueError('Load weights is not yet supported with TPUStrategy '
    249                          'with steps_per_run greater than 1.')
--> 250     return super(Model, self).load_weights(filepath, by_name, skip_mismatch)
    251 
    252   def compile(self,

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/tensorflow/python/keras/engine/network.py in load_weights(self, filepath, by_name, skip_mismatch)
   1229     else:
   1230       try:
-> 1231         py_checkpoint_reader.NewCheckpointReader(filepath)
   1232         save_format = 'tf'
   1233       except errors_impl.DataLossError:

~/opt/miniconda2/envs/AutoML/lib/python3.8/site-packages/tensorflow/python/training/py_checkpoint_reader.py in NewCheckpointReader(filepattern)
     93   """
     94   try:
---> 95     return CheckpointReader(compat.as_bytes(filepattern))
     96   # TODO(b/143319754): Remove the RuntimeError casting logic once we resolve the
     97   # issue with throwing python exceptions from C++.

ValueError: Unsuccessful TensorSliceReader constructor: Failed to get matching files on ./structured_data_regressor/trial_a2b0718070dcd1d815fe093a8ebb90ab/checkpoints/epoch_52/checkpoint: Not found: ./structured_data_regressor/trial_a2b0718070dcd1d815fe093a8ebb90ab/checkpoints/epoch_52; No such file or directory

**Comment by achaar** [comment]
Hello @KirkDCO , Can you please try again using the method mentioned in : https://github.com/achaar/autokeras/blob/img_reg_tutorial/docs/templates/tutorial/image_regression.ipynb

Please update the AutoKeras-tuner also.

Thanks.

**Comment by KirkDCO** [comment]
I'm not sure I understand what method you're referring to.   The code you linked uses an ImageRegressor and is very focused around that concept instead of a StructuredDataRegression.

Regarding updating AutoKeras-tuner, are you referring to keras-tuner (https://github.com/keras-team/keras-tuner)?  The example pulls from a specific commit which seems very unmaintainable.

**Comment by achaar** [comment]
Yes, you can use the specific commit for the time being. The issue will be resolved in the next release. 

I was referring to the installation procedure for AutoKeras and keras-tuner.

**Comment by achaar** [comment]
Okay, then we'll try to resolve this and get back as soon as possible.

**Comment by pingusix** [comment]
I am seeing this as well with structured_data_classifier after trying the various combinations of code mentioned earlier.  I get the error a lot of the time though not always and when it fails I can verify that the checkpoint  does not exist under the trials_xx/checkpoints/epoch_yy path. The directory parameter is specified as an absolute path and grep doesn't seem to show that the file has been saved somewhere else as far as I can see.

**Comment by chyt** [comment]
I am seeing this as well using ImageClassifier on Google Colab connected to Google Drive. Usually everything is fine with a smaller number of trials/epochs. However when I increase this number (right now I am doing 10 trials and 200 epochs) I get the error and the file does not exist in the directory. 

I was wondering if it could have something to do with getting disconnected from Google Colab, but based on the reports here it seems like it is not the case. 

I'm using autokeras 1.0.3, keras tuner 1.0.2rc0, and tensorflor 2.2.0.

**Comment by haifeng-jin** [comment]
Originally, this error is because Keras Tuner deletes the old checkpoints saved on disk to reduce the disk usage. However, we have fixed this in a recent pull request (https://github.com/keras-team/keras-tuner/pull/318). Not sure why still exists.

@chyt Would you share a colab notebook for the reproduction? Thanks.

**Comment by chyt** [comment]
> Originally, this error is because Keras Tuner deletes the old checkpoints saved on disk to reduce the disk usage. However, we have fixed this in a recent pull request ([keras-team/keras-tuner#318](https://github.com/keras-team/keras-tuner/pull/318)). Not sure why still exists.
> 
> @chyt Would you share a colab notebook for the reproduction? Thanks.

How would I share this notebook with you? The code is pretty straightforward, but the dataset is loaded from Google Drive.

**Comment by mreismendes** [comment]
I've been having the same problem quoted by @chyt!
![image](https://user-images.githubusercontent.com/32521301/86279654-c27f9200-bbb0-11ea-974e-01d968e49b7a.png)

**Comment by haifeng-jin** [comment]
@chyt It seems you are facing multiple bugs while using AutoKeras. It would be great if we can schedule an one hour meeting to help you debug. We can use half of the meeting to debug and half of the meeting for you to answer some of our user study questions. You can join our slack and message me your email address. Thank you!

**Comment by ricwo** [comment]
@haifeng-jin I'm having the same issue. ak is trying to restore epoch 2, which was not persisted:

```bash
ValueError: Unsuccessful TensorSliceReader constructor: Failed to get matching files on /data/auto-clf-20-epochs-4-trials/trial_479975cd710064c8120d0f95a2c0b230/checkpoints/epoch_2/checkpoint: Not found: /data/auto-clf-20-epochs-4-trials/trial_479975cd710064c8120d0f95a2c0b230/checkpoints/epoch_2; No such file or directory
```
However epoch 2 was not persisted during training:
```bash
ls /data/auto-clf-20-epochs-4-trials/trial_479975cd710064c8120d0f95a2c0b230/checkpoints
epoch_0  epoch_10  epoch_11  epoch_12  epoch_3  epoch_4  epoch_5  epoch_6  epoch_7  epoch_8  epoch_9
```

Is there anything I can do to help fix this issue?

**Comment by haifeng-jin** [comment]
@ricwo Are you using overwrite=False? If you let overwrite=True, I think the error would gone.

**Comment by ricwo** [comment]
@haifeng-jin setting `overwrite=False` indeed fixed the issue. Thanks for your help!

**Comment by sunseaandpalms** [comment]
Is `overwrite=True` not working yet? Any stable solution to continue the seeking/training after being stopped?

**Comment by haifeng-jin** [comment]
We have fixed the problem in the master branch. You can use the master branch with keras-tuner 1.0.2rc1 tag.

**Comment by jacobswain** [comment]
I'm on 1.0.2rc1 and I'm getting the same error

**Comment by haifeng-jin** [comment]
We have just had a new release 1.0.5. I expect use it with TF 2.3.0 can fix the bug. We have fixed another bug in that might cause the problem so it should work this time.

**Comment by swapnil96** [comment]
Getting error
`ValueError: Unsuccessful TensorSliceReader constructor: Failed to get matching files on ./structured_data_regressor/trial_869518eb879f942102cf397ccdff8288/checkpoints/epoch_2/checkpoint: Not found: ./structured_data_regressor/trial_869518eb879f942102cf397ccdff8288/checkpoints/epoch_2; No such file or directory`

`ls structured_data_regressor/trial_869518eb879f942102cf397ccdff8288/checkpoints/`
`epoch_0  epoch_10  epoch_11  epoch_12  epoch_3  epoch_4  epoch_5  epoch_6  epoch_7  epoch_8  epoch_9`


>>> import autokeras as ak
>>> ak.__version__
'1.0.8'
>>> import tensorflow as tf
>>> tf.__version__
'2.3.0'
>>> import kerastuner
>>> kerastuner.__version__
'1.0.2rc1'

**Comment by q-55555** [comment]
Same problem on my side : tensorflow 2.3.1, autokeras 1.0.9, keras-tuner 1.0.2rc2.
Impossible to use autokeras... Any workaround ?

**Comment by haifeng-jin** [comment]
I cannot reproduce the error. Is there any colab examples that reproduces the error? Thanks.

**Comment by q-55555** [comment]
Here is the colab to reproduce the error:
https://colab.research.google.com/drive/1jOzTuL26UZaISnSZkUD273yZ1cqrX1QL?usp=sharing

I think the error appears when the number of epochs is quite high (500 in the example). 
I have the impression that only 10 checkpoints are saved in the file system and when autokeras tries to load a previous checkpoint it crashes.

**Comment by mh-hassan18** [comment]
Facing the same problem, here are the version details:
tensorflow 2.3.1
autokeras 1.0.9
keras-tuner 1.0.2rc2

**Comment by haifeng-jin** [comment]
The fix is in a pending pull request to keras tuner.
https://github.com/keras-team/keras-tuner/pull/424
We will have a new tag for kerastuner after this one is merged.

**Comment by mh-hassan18** [comment]
How long will it take to fix the bug?

**Comment by haifeng-jin** [comment]
This bug has been fixed in Keras Tuner 1.0.2.

**Comment by slimebob1975** [comment]
Encountered a similar error in tensorflow 2.12.0 using scikeras example code:

[https://github.com/adriangb/scikeras/issues/291](url)
